indexer/src/main/java/config/ConsumerConfiguration.java

```diff
@@ -0,0 +1,24 @@
+package config;
+
+import java.util.Map;
+
+public class ConsumerConfiguration {
+    private final String HOST_KEY = "host";
+    private final String PORT_KEY = "port";
+    private final String GROUP_ID_KEY = "group.id";
+
+    private String host;
+    private int port;
+    private String groupId;
+
+    public ConsumerConfiguration(Map<String, Object> jsonMap) {
+        this.host = (String) jsonMap.get(HOST_KEY);
```

Comment:
Let's add `requireNonNull` for every configuration param which you think is strictly required

Reply:
as agreed, i will change ConfigurationFactory s.t. configuration classes will be created using jackson. Using @NotNull where the configuration is created will prevent null members.